mne_bids_pipeline/steps/preprocessing/_08_ptp_reject.py

```diff
@@ -79,46 +79,75 @@ def drop_ptp(
 
     # Get rejection parameters and drop bad epochs
     epochs = mne.read_epochs(in_files.pop("epochs"), preload=True)
-    reject = _get_reject(
-        subject=subject,
-        session=session,
-        reject=cfg.reject,
-        ch_types=cfg.ch_types,
-        param="reject",
-        epochs=epochs,
-    )
-    if cfg.spatial_filter == "ica":
-        ica_reject = _get_reject(
+
+    if cfg.reject == "autoreject_local":
+        import numpy as np
+        import autoreject
+
+        msg = "Using autoreject to find and repair bad epochs"
+        logger.info(**gen_log_kwargs(message=msg))
+
+        ar = autoreject.AutoReject(
+            n_interpolate=np.array(cfg.autoreject_n_interpolate),
+            random_state=cfg.random_state,
+            n_jobs=exec_params.n_jobs,
+            verbose=False,
+        )
+        n_epochs_before_reject = len(epochs)
+        epochs, reject_log = ar.fit_transform(epochs, return_log=True)
+        n_epochs_after_reject = len(epochs)
+        assert (
+            n_epochs_before_reject - n_epochs_after_reject
+            == reject_log.bad_epochs.sum()
+        )
+
+        msg = (
+            f"autoreject marked {reject_log.bad_epochs.sum()} epochs as bad "
+            f"(cross-validated n_interpolate limit: {ar.n_interpolate_})"
+        )
+        logger.info(**gen_log_kwargs(message=msg))
+    else:
+        reject = _get_reject(
```

Comment:
The diff doesn't look nice here, but I basically just added this new `if` block and moved the previous code into the `else` section